File: amp-cli/src/commands/show.ts
import { Command } from "commander";
import { requireNodeById } from "../core/node.js";
import { readGraphIndex } from "../core/index.js";
import { requireStore } from "../core/store.js";
import { chalk, colorStatus, colorType, dim, heading, kv } from "../utils/ui.js";

interface ShowOptions {
  json?: boolean;
}

export function registerShowCommand(program: Command): void {
  program
    .command("show <id>")
    .description("Display a node's full content and metadata")
    .option("--json", "Output raw JSON", false)
    .action((id: string, options: ShowOptions) => {
      const paths = requireStore(process.cwd());
      const node = requireNodeById(paths, id);

      if (options.json) {
        console.log(JSON.stringify({ frontmatter: node.frontmatter, content: node.content }, null, 2));
        return;
      }

      const fm = node.frontmatter;
      console.log(heading(fm.id));
      console.log(kv("type", colorType(fm.type)));
      console.log(kv("status", colorStatus(fm.status)));
      console.log(kv("created", fm.created));
      console.log(kv("modified", fm.modified));
      console.log(kv("author", fm.author));
      if (fm.source) console.log(kv("source", fm.source));
      if (fm.scope) console.log(kv("scope", fm.scope));
      if (fm.confidence !== undefined) console.log(kv("confidence", String(fm.confidence)));
      if (fm.tags?.length) console.log(kv("tags", fm.tags.join(", ")));
      if (fm.superseded_by) console.log(kv("superseded_by", fm.superseded_by));
      if (fm.ttl) console.log(kv("ttl", fm.ttl));
      if (fm.links?.length) {
        console.log(kv("links", fm.links.map((l) => `${l.target} (${l.relation})`).join(", ")));
      }

      const graph = readGraphIndex(paths);
      const backlinks = graph?.nodes[fm.id]?.incoming ?? [];
      if (backlinks.length) {
        console.log(kv("backlinks", backlinks.map((l) => `${l.target} (${l.relation})`).join(", ")));
      }

      console.log();
      console.log(chalk.dim("─".repeat(60)));
      console.log(node.content);
      console.log(chalk.dim("─".repeat(60)));
      console.log(dim(node.filePath));
    });
}

File: amp-python/src/amp/commands/show.py
from __future__ import annotations

import click
from rich.markdown import Markdown
from rich.panel import Panel
from rich.table import Table

from ._common import console, get_store, handle_amp_errors, status_badge, type_badge


@click.command()
@click.argument("node_id")
@click.pass_context
@handle_amp_errors
def show_cmd(ctx: click.Context, node_id: str) -> None:
    """Display a node's full content and metadata."""
    store = get_store(ctx.obj.get("path"))
    node = store.show(node_id)

    meta = Table.grid(padding=(0, 2))
    meta.add_column(style="bold")
    meta.add_column()
    meta.add_row("id", node.id)
    meta.add_row("type", type_badge(node.type))
    meta.add_row("status", status_badge(node.status))
    meta.add_row("author", node.author)
    meta.add_row("created", node.created)
    meta.add_row("modified", node.modified)
    if node.source:
        meta.add_row("source", node.source)
    if node.confidence is not None:
        meta.add_row("confidence", str(node.confidence))
    if node.scope:
        meta.add_row("scope", node.scope)
    if node.tags:
        meta.add_row("tags", ", ".join(node.tags))
    if node.superseded_by:
        meta.add_row("superseded_by", node.superseded_by)
    if node.links:
        meta.add_row("links", ", ".join(f"{l.target} ({l.relation})" for l in node.links))
    meta.add_row("path", node.path or "-")

    console.print(Panel(meta, title=f"[bold]{node.id}[/bold]", border_style="cyan"))
    console.print(Markdown(node.content))
